python/ray/data/_internal/progress_bar.py

```diff
@@ -22,6 +24,60 @@
 _canceled_threads_lock = threading.Lock()
 
 
+@ray.remote
+class _ProgressBarRouter:
+    def __init__(self):
+        self.commands = []
+
+    async def submit_command(self, cmd):
+        self.commands.append(cmd)
+
+    async def fetch_commands(self):
+        out = []
+        while not out:
+            out = self.commands.copy()
+            self.commands.clear()
+            if not out:
+                await asyncio.sleep(0.1)
+        return out
+
+
+class _ProgressTailer(threading.Thread):
+    def __init__(self, bar):
+        self.bar = bar
+        self.progress_bars = {}  # Map from uuid to ProgressBar object.
+        super().__init__(daemon=True)
+
+    def run(self):
+        while True:
+            try:
+                cmds = ray.get(self.bar.fetch_commands.remote())
+                for bar_uuid, method, args in cmds:
+                    if method == "__init__":
+                        self.progress_bars[bar_uuid] = ProgressBar(
+                            *args, _is_central=True
+                        )
+                    else:
+                        bar = self.progress_bars[bar_uuid]
+                        getattr(bar, method)(*args)
+            except Exception as e:
+                print(e)
+
+
+@PublicAPI(stability="beta")
+def enable_centralized_progress_bar() -> None:
```

Comment:
Another possibility is to record progress in the Ray logs, and using LogMonitor to route/parse progress events for rendering.